lib/shimmer.js

```diff
@@ -579,6 +576,60 @@ function instrumentPostLoad(agent, nodule, moduleName, resolvedName, esmResolver
   return nodule
 }
 
+/**
+ * If a module that is being shimmed was loaded via a typical `require` then
+ * we can use it as normal. If was loaded via an ESM import, then we need to
+ * handle it with some extra care. This function inspects the module,
+ * the conditions around its loading, and returns an appropriate object for
+ * subsequent shimming methods.
+ *
+ * @param {object} params Input parameters.
+ * @param {object} params.nodule The nodule being instrumented.
+ * @param {object} params.instrumentation The configuration for the nodule
+ * to be instrumented.
+ * @param {boolean|null} params.esmResolver Indicates if the nodule was loaded
+ * via an ESM import.
+ * @returns {object} The nodule or a Proxy.
+ */
+function resolveNodule({ nodule, instrumentation, esmResolver }) {
+  if (instrumentation.isEsm === true || !esmResolver) {
+    return nodule
+  }
+
+  // We have a CJS module wrapped by import-in-the-middle having been
+  // imported through ESM syntax. Due to the way CJS modules are parsed by
+  // ESM's import, we can have the same "export" attached to the `default`
+  // export and as a top-level named export. In order to shim things such
+  // that our users don't need to know to access `something.default.foo`
+  // when they have done `import * as something from 'something'`, we need
+  // to proxy the proxy in order to set our wrappers on both instances.
+  const noduleDefault = nodule.default
+  const origNodule = nodule
+  return new Proxy(
```

Comment:
this is the root of the fix but the conventional commit is very vague and technically untrue.  This fixes all cjs modules and instrumenting the right thing depending on the import style correct?

Reply:
Yes, that is an accurate assessment. I must admit that I'm not well versed in conventional commit. What message do you think would be more appropriate?